src/hooks/__internal__/useCharacterCount/useCharacterCount.tsx

```diff
@@ -16,7 +16,13 @@ const useCharacterCount = (
   const isCharacterLimitValid =
     typeof characterLimit === "number" && !Number.isNaN(characterLimit);
   const l = useLocale();
-  const hintString = l.characterCount.hintString();
+
+  const getFormatNumber = (rawValue: number, locale: string) =>
+    new Intl.NumberFormat(locale).format(rawValue);
+
+  const hintString = l.characterCount.hintString(
+    getFormatNumber(characterLimit as number, l.locale())
+  );
```

Comment:
**issue**: I think we need a check here in case `characterLimit` is not undefined rather than doing a type assertion, since if `characterLimit` is undefined, there is a chance the code on line 21 could throw an error.

```tsx
const hintString =
     isCharacterLimitValid 
        ? l.characterCount.hintString(getFormatNumber(characterLimit, l.locale())
        : undefined;
```

Reply:
We already have a similar check in the file on line 45